python/pyspark/ml/classification.py

```diff
@@ -1630,8 +1614,52 @@ def _to_java(self):
         _java_obj.setPredictionCol(self.getPredictionCol())
         return _java_obj
 
+    def _make_java_param_pair(self, param, value):
```

Comment:
I'm not a big fan of copying these methods.  Does it work to use the parent versions of this method and _transfer_param_map_to_java if you set self._java_obj in this class' __init__ method?  I guess you'll have to override _transfer_param_map_from_java, but 1 is better than 3.

Note for other reviewers: This is a short/medium-term fix to permit persistence, but we'll change it eventually to permit Python-only base estimators.